src/field/pokemon.ts

```diff
@@ -3586,6 +3587,51 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return false;
   }
 
+  /**
+   * Helper function that determines if a Pokemon has a non-volatile status effect and/or is Confused
+   * @param statusList the status(es) to be checked
+   * @param includeConfusion whether Confusion should also be considered
+   * @param ignoreAbility whether a status effect-mocking ability should be considered
+   * @returns `true` if the Pokemon has a status effect | `false` if it does not
+   */
+  hasStatusEffect(
+    statusList: StatusEffect | StatusEffect[],
+    includeConfusion: boolean = false,
+    ignoreAbility: boolean = false,
+  ): boolean {
+    const result = new BooleanHolder(false);
+    if (!Array.isArray(statusList)) {
+      statusList = [statusList];
+    }
+    if (!ignoreAbility) {
+      applyAbAttrs(MockStatusEffectAbAttr, this, false, result, statusList);
+    }
+    if (
+      (!!this.status && statusList.includes(this.status.effect))
```

Comment:
```suggestion
      (this.status && statusList.includes(this.status.effect))
```
You don't need the `!!` in an `if` block, only in the `() => ...` functions because those don't force values to be evaluated as booleans.